jax/_src/ops/scatter.py

```diff
@@ -15,11 +15,21 @@
 # Helpers for indexed updates.
 
 
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+
 from jax import lax
 from jax._src.numpy import lax_numpy as jnp
 from jax._src import util
 
 
+Index = Union[int, slice, Sequence[int], np.ndarray, 'ellipsis',
+              Tuple[Union[int, slice, Sequence[int], np.ndarray, 'ellipsis'], ...]]
+Scalar = Union[float, int]
+Numeric = Union[jnp.ndarray, Scalar]
```

Comment:
Type checking `jnp.ndarray` doesn't actually work with mypy at the moment, I think. We get around this elsewhere by aliasing `Array = Any` and using that instead.

Reply:
Sure, let's do that.  I was just copying some of the other files that are annotating with `jnp.ndarray`.

In any case, it appears that there are circular imports because `_src/numpy/lax_numpy.py` import `ops`, which imports `_src/ops/scatter.py`, and something will have to be done about that.  For example, `ndarray` could be moved to its own file.

I'll update to Array = Any as you suggest.

Reply:
Hmm... I'm not sure. From what I understand, variables returned from a function annotated with `jnp.ndarray` will not typecheck correctly further in the program because, for example, jnp.ndarray doesn't define all the attributes and methods that are available on a JAX DeviceArray. It may be that nobody has hit this because nobody has tried using these functions in annotated code. But I suppose if it's not causing people issues it's not worth spending time to fix!

Reply:
Okay, we'll leave it then in the other file if that's what you think is best. :smile: